ast/desugar/Desugar.cc

```diff
@@ -308,10 +310,12 @@ unique_ptr<Expression> desugarMlhs(DesugarContext dctx, core::Loc loc, parser::M
     }
 
     auto expanded = MK::Send3(loc, MK::Constant(loc, core::Symbols::Magic()), core::Names::expandSplat(),
-                              std::move(rhs), MK::Int(loc, before), MK::Int(loc, after));
-    stats.insert(stats.begin(), MK::Assign(loc, tempName, std::move(expanded)));
+                              MK::Local(loc, tempRhs), MK::Int(loc, before), MK::Int(loc, after));
+    // space reserved above
+    stats[0] = MK::Assign(loc, tempRhs, std::move(rhs));
+    stats[1] = MK::Assign(loc, tempExpanded, std::move(expanded));
 
-    return MK::InsSeq(loc, std::move(stats), MK::Local(loc, tempName));
+    return MK::InsSeq(loc, std::move(stats), MK::Local(loc, tempRhs));
```

Comment:
It took me a bit playing around with this locally before I understood it (because I was not familiar with `Magic.<expandSplat>` previously). One comment which would have helped me understand this better would have been this:

```suggestion
    // Regardless of how we destructure an assignment, Ruby evaluates the expression to the entire right hand side,
    // not any individual component of the destructured assignment.
    return MK::InsSeq(loc, std::move(stats), MK::Local(loc, tempRhs));
```